### PR TITLE
Add error type in NSXApiError

### PR DESCRIPTION
**Pull request by TaoZou1**
Operator sometimes will check error type instead of just returning error. 
Add error type in NSXApiError

### CODE REVIEW

File: pkg/nsx/services/nsxserviceaccount/cluster.go [hunk 1]
```diff
@@ -196,10 +196,10 @@ func (s *NSXServiceAccountService) RestoreRealizedNSXServiceAccount(ctx context.
...
 	}
-	switch err.(type) {
-	case vapierrors.NotFound:
-	default:
-		return err
+	if nsxErr, ok := err.(*nsxutil.NSXApiError); ok {
+		if nsxErr.Type() != vapierrors.ErrorType_NOT_FOUND {
+			return err
+		}
 	}
```
Comment (gran-vmv) [inline pkg/nsx/services/nsxserviceaccount/cluster.go:203]: Also `else {return err}`?

Comment (TaoZou1) [reply pkg/nsx/services/nsxserviceaccount/cluster.go:205]: done.

**Review by zhengxiexie — APPROVED** [review]
LGTM

**Review by gran-vmv — APPROVED** [review]

**Pull request merged**